python/ray/tests/test_client_multi.py

```diff
@@ -0,0 +1,173 @@
+import sys
+import pytest
+import ray
+
+
+@pytest.mark.skipif(
+    sys.platform == "win32",
+    reason="PSUtil does not work the same on windows.")
+@pytest.mark.parametrize(
+    "call_ray_start",
+    ["ray start --head --ray-client-server-port 25001 --port 0"],
+    indirect=True)
+def test_multi_cli_basic(call_ray_start):
+    cli1 = ray.init("ray://localhost:25001", allow_multiple=True)
+    cli2 = ray.init("ray://localhost:25001", allow_multiple=True)
+    with cli1:
+        a = ray.put(10)
+    with cli2:
+        b = ray.put(20)
+
+    with pytest.raises(Exception):
+        ray.put(30)
+
+    with pytest.raises(Exception):
+        ray.get(a)
+
+    with pytest.raises(Exception), cli2:
+        ray.get(a)
+
+    with pytest.raises(Exception), cli1:
+        ray.get(b)
+
+    ray.init("ray://localhost:25001")
```

Comment:
Hmm in retrospect this is a bit odd. Maybe it should also raise an error (can fix later).